basic_autonomy_ros2/CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+
+# Copyright (C) 2022 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+cmake_minimum_required(VERSION 3.5)
+project(basic_autonomy_ros2)
+
+# Declare carma package and check ROS version
+find_package(carma_cmake_common REQUIRED)
+carma_check_ros_version(2)
+carma_package()
+
+## Find dependencies using ament auto
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
+
+# Name build targets
+set(node_lib basic_autonomy_ros2_lib)
+
+# Includes
+include_directories(
+  include
+)
+
+# Build
+ament_auto_add_library(${node_lib} SHARED
+        src/basic_autonomy.cpp
+        src/helper_functions.cpp
+        src/log/log.cpp
+        src/smoothing/BSpline.cpp
+        src/smoothing/filters.cpp
+)
+
+target_link_libraries(
+        ${node_lib}
+        ${Boost_LIBRARIES}
```

Comment:
Is this line needed? Should there be find_package for Boost as well or adding it to the package.xml?